dipy/tracking/tests/test_utils.py

```diff
@@ -224,48 +223,62 @@ def test_target():
+    _target(target, streamlines, (0, 0, 0), (1, 0, 0), True)
+
+
+def test_target_lb():
+    streamlines = [np.array([[0., 1., 1.],
+                             [3., 1., 1.]]),
+                   np.array([[0., 0., 0.],
+                             [2., 2., 2.]])]
+    _target(target_line_based, streamlines, (1, 1, 1), (2, 1, 1), False)
+
+
+def _target(target_f, streamlines, voxel_both_true, voxel_one_true,
+            test_bad_points):
     affine = np.eye(4)
     mask = np.zeros((4, 4, 4), dtype=bool)
-    mask[0, 0, 0] = True
 
     # Both pass though
-    new = list(target(streamlines, mask, affine=affine))
+    mask[voxel_both_true] = True
+    new = list(target_f(streamlines, mask, affine=affine))
     assert_equal(len(new), 2)
-    new = list(target(streamlines, mask, affine=affine, include=False))
+    new = list(target_f(streamlines, mask, affine=affine, include=False))
     assert_equal(len(new), 0)
 
     # only first
     mask[:] = False
-    mask[1, 0, 0] = True
-    new = list(target(streamlines, mask, affine=affine))
+    mask[voxel_one_true] = True
+    new = list(target_f(streamlines, mask, affine=affine))
     assert_equal(len(new), 1)
     assert_true(new[0] is streamlines[0])
-    new = list(target(streamlines, mask, affine=affine, include=False))
+    new = list(target_f(streamlines, mask, affine=affine, include=False))
     assert_equal(len(new), 1)
     assert_true(new[0] is streamlines[1])
 
     # Test that bad points raise a value error
-    bad_sl = [np.array([[10., 10., 10.]])]
-    new = target(bad_sl, mask, affine=affine)
-    assert_raises(ValueError, list, new)
-    bad_sl = [-np.array([[10., 10., 10.]])]
-    new = target(bad_sl, mask, affine=affine)
-    assert_raises(ValueError, list, new)
+    if test_bad_points:
+        bad_sl = streamlines + [np.array([[10.0, 10.0, 10.0]])]
+        new = target_f(bad_sl, mask, affine=affine)
+        assert_raises(ValueError, list, new)
+        bad_sl = streamlines + [-np.array([[10.0, 10.0, 10.0]])]
+        new = target_f(bad_sl, mask, affine=affine)
+        assert_raises(ValueError, list, new)
 
     # Test smaller voxels
     affine = np.random.random((4, 4)) - .5
     affine[3] = [0, 0, 0, 1]
     streamlines = list(move_streamlines(streamlines, affine))
-    new = list(target(streamlines, mask, affine=affine))
+    new = list(target_f(streamlines, mask, affine=affine))
     assert_equal(len(new), 1)
     assert_true(new[0] is streamlines[0])
-    new = list(target(streamlines, mask, affine=affine, include=False))
+    new = list(target_f(streamlines, mask, affine=affine, include=False))
     assert_equal(len(new), 1)
     assert_true(new[0] is streamlines[1])
 
-    # Test that changing mask and affine do not break target
-    include = target(streamlines, mask, affine=affine)
-    exclude = target(streamlines, mask, affine=affine, include=False)
+    # Test that changing mask and affine do not break target_line_based
```

Comment:
In fact, this tests that it doesn,t break either target function, since the current function is called for both target versions.